### PR TITLE
Arm32 support

### PR DESCRIPTION
**Pull request by cpriouzeau**
- Patch of EDK2 for ARM32 support
- Adaptation of list of BBR SCT

### CODE REVIEW

**Comment by etienne-lms** [comment]
Hello @cpriouzeau and all,
Base on this P-R, i've created https://github.com/ARM-software/bbr-acs/pull/9 and  https://github.com/ARM-software/bbr-acs/pull/10, as proposals for discussions around 32bit architecture support in bbr-acs.

P-R #9 targets a common BBR_SCT.dsc file for AARCH64 and ARM.
P-R #10 targets a common build_bbr.sh script and prevents need for your commit "ARM32 support: EDK2: patch for ARM32 support".
Dear maintainers, do these changes make sense to you?

These 2 P-Rs do not cover all required changes in other build scripts (`build-sct.sh`, `buildf-wts.sh`, `build-uefi-apps.sh`) for toolchain/architecture selection. I think these could be factorized with related changes in https://github.com/ARM-software/arm-systemready/pull/25.

File: common/scripts/build-sct.sh [hunk 1]
```diff
@@ -69,6 +67,24 @@ if ! [[ $BUILD_PLAT = IR ]] && ! [[ $BUILD_PLAT = ES ]] ; then
     exit
 fi
 
+. $TOP_DIR/../../common/config/common_config.cfg
```
Comment (edhay) [inline common/scripts/build-sct.sh:70]: Please refactor the changes in file to a new file common/scripts/build-sct-arm.sh

The original file common/scripts/build-sct.sh shall be unmodified.

File: common/scripts/build-uefi-apps.sh [hunk 2]
```diff
@@ -62,6 +63,20 @@ if [ $BUILD_PLAT = SR ]; then
    BUILD_PLAT=ES
 fi
```
Comment (edhay) [inline common/scripts/build-uefi-apps.sh:65]: Please refactor the changes in file to a new file common/scripts/build-uefi-apps-arm.sh

The original file common/scripts/build-uefi-apps.sh shall be unmodified.

**Review by edhay — CHANGES_REQUESTED** [review]
Please check my comments on refactoring. This is to minimize the impact on the 64-bit ACS

File: common/scripts/build-uefi-apps.sh [hunk 3]
```diff
@@ -95,7 +110,25 @@ do_build()
...
 
     export EDK2_TOOLCHAIN=$UEFI_TOOLCHAIN
-    export ${UEFI_TOOLCHAIN}_AARCH64_PREFIX=$CROSS_COMPILE
+    if [[ ! ${ARCH+x} ]]; then
+            # use ARCH=arm64
+            CROSS_COMPILE_DIR=$(dirname $CROSS_COMPILE)
+            PATH="$PATH:$CROSS_COMPILE_DIR"
+            export ${UEFI_TOOLCHAIN}_AARCH64_PREFIX=$CROSS_COMPILE
+            ARCH_TARGET_FOR_UEFI=AARCH64
+        else
```
Comment (vstehle) [inline common/scripts/build-uefi-apps.sh:119]: Indentation does not seem to match the opening if.

**Comment by etienne-lms** [comment]
Can close this P-R as superseded by https://github.com/ARM-software/bbr-acs/pull/11.

**Pull request closed without merging**